ffi/device.cc

```diff
@@ -20,36 +20,46 @@ void init_ffi_device(py::module_ &m) {
             [](const Ref<Target> &target, bool useNativeArch) {
                 target->setUseNativeArch(useNativeArch);
             },
-            "useNativeArch"_a = true)
+            "use_native_arch"_a = true)
         .def("use_native_arch", &Target::useNativeArch)
         .def("__str__", &Target::toString)
         .def("main_mem_type", &Target::mainMemType)
         .def("__eq__",
              static_cast<bool (*)(const Ref<Target> &, const Ref<Target> &)>(
-                 &isSame));
+                 &isSameTarget));
+
     py::class_<CPU, Ref<CPU>>(m, "CPU", pyTarget)
         .def(py::init([](bool useNativeArch) {
                  return Ref<CPU>::make(useNativeArch);
              }),
              "use_native_arch"_a = true);
+
+#ifndef FT_WITH_CUDA
     py::class_<GPU, Ref<GPU>>(m, "GPU", pyTarget)
         .def(py::init([](bool useNativeArch) {
                  return Ref<GPU>::make(useNativeArch);
              }),
+             "use_native_arch"_a = true);
```

Comment:
Since a non-empty GPU `Target` can only be constructed from a `Device` now, we can remove the support of constructing a `Target` from Python. The name of `CPU` and `GPU` in Python side can be reused as helper functions to construct `Device(TargetType.CPU, ...)` and `Device(TargetType.GPU, ...)`.